feat: add docs on Apify store tiered discount pricing options (#1572)
apify/apify-docs: sources/platform/actors/publishing/monetize.mdx

3. **Pay per result (PPR)**: Users don't pay for the platform usage costs. Instead, they pay the developer based on the number of results produced by the Actor.
4. **Pay per event (PPE)**: Users don't pay for the platform usage cost the Actor generates. Instead, they pay based on specific events that are programmatically triggered from the Actor's source code. These events are defined by the developer and can include actions such as generating a single result or starting an Actor.

> For a detailed comparison of pricing models from the perspective of your users, refer to our [user-facing documentation](/platform/actors/running/actors-in-store).

Reviewer: This page is also `our user-facing documentation`, so this sentence makes little sense.

```suggestion
For a detailed comparison of pricing models from the perspective of your users, refer to [Actors in Store](/platform/actors/running/actors-in-store) page.
```

> ### Pay-per-result pricing model

Reviewer 2: ```suggestion
### Pay-per-result pricing model (PPR)
```

Reviewer 2: Let's change it in the PPE heading as well. I can't do it, because it's too far away from edited lines :D

### Computing your costs for PPE and PPR Actors
For both PPE and PPR Actors, profit is computed using the formula `(0.8 * revenue) - costs`. In this section, we'll explain how the `costs` component is calculated.

When paying users run your Actor, it generates platform usage in the form of compute units, data traffic, API operations etc. This usage determines the `costs` in the profit formula above. Note that usage by free users is covered by Apify and doesn't affect your costs. To calculate costs in dollars, multiply the unit cost of each service by the quantity consumed. For example, if a run uses 10 compute units (CUs) at $0.5/CU, your cost would be $5.

As mentioned in the [How to attract larger customers of PPE and PPR Actors](#how-to-attract-larger-customers-of-ppe-and-ppr-actors) section, the user's discount tier determines the unit costs applied to their runs if the Actor has tiered discounts set up. Higher tiers have lower unit costs, allowing you to offer lower prices to high-tier customers while maintaining healthy profit margins.

The table below summarizes the unit prices for cost computation across different discount tiers.

> | Service                         | Price is per | `FREE`  | `BRONZE` | `SILVER` | `GOLD`  | `PLATINUM` |

Reviewer 2: I thought we agreed that we won't show the platinum pricing. Did I understand it wrong? Or maybe this was done before we talked?

Author: Yeah, this was done before, but I just haven't had time to update it yet. My apologies, will re-request review once updated.

| SERPs proxy                    | 1,000 SERPs   | $2.5    | $2.5     | $2       | $1.7    | $1.25      |
| Data transfer - external       | GB            | $0.2    | $0.2     | $0.19    | $0.18   | $0.18      |
| Data transfer - internal       | GB            | $0.05   | $0.05    | $0.045   | $0.04   | $0.04      |
| Dataset - reads               | 1,000 reads   | $0.0004 | $0.0004  | $0.00036 | $0.00032| $0.00032   |
| Dataset - writes              | 1,000 writes  | $0.005  | $0.005   | $0.0045  | $0.004  | $0.004     |
| Key-value store - reads       | 1,000 reads   | $0.005  | $0.005   | $0.0045  | $0.004  | $0.004     |
| Key-value store - writes      | 1,000 writes  | $0.05   | $0.05    | $0.045   | $0.04   | $0.04      |
| Key-value store - lists       | 1,000 lists   | $0.05   | $0.05    | $0.045   | $0.04   | $0.04      |
| Request queue - reads         | 1,000 reads   | $0.004  | $0.004   | $0.0036  | $0.0032 | $0.0032    |
| Request queue - writes        | 1,000 writes  | $0.02   | $0.02    | $0.018   | $0.016  | $0.016     |

If you decide not to offer tiered discounts on your Actor, the unit prices for FREE tier apply.

> 

Reviewer 2: I think it would also make sense to add a section on using high Free pricing to limit usage for free users. It's one of the hidden benefits of the tiered pricing as well, that you can now customize Free pricing individually. Let me know if I should elaborate more.

Author: I actually considered whether we should explicitly advise this :D. At the end, I decided that the developers will do that regardless, even without our advice. But if you think it's a good idea, I'll add it there explicitly.

Reviewer 2: Yeah, I think it will end up becoming the "best" practice even if we don't document it, so I'd get ahead of it and document it. Perhaps add something like:

While we don't recommend setting high prices for Free users by default, because it limits the users' ability to evaluate your Actor, we understand that sometimes it's necessary to protect you from fraud or excessive use of your internal APIs. For situations like this we recommend ....

:::

### How to attract larger customers of PPE and PPR Actors

> Each user running your Actor belongs to a specific Apify Store discount tier, namely `FREE`, `BRONZE`, `SILVER`, or `GOLD` tier.

Reviewer: I think emphasis works better than inline code, also shouldn't we mentioned _DIAMOND_ & _PLATINUM_ already here since it is a possible tier?

```suggestion
Each user running your PPE or PPR Actor belongs to a discount tier:
- _FREE_
- _BRONZE_
- _SILVER_
- _GOLD_

```

Author: @TC-MO we agreed with @mnmkng that diamond and platinum are kind of hidden and won't be actively advertised.

Reviewer: That does make sense but both of these tiers are mentioned on line 143. I would advise two solutions:

1. Remove all mentions of those if they are not to be adverstised in user-facing content
2. Add them here like so to explain that, yes they exist, but are specially reserved only for a subset of customers

```
Each user running your PPE or PPR Actor belongs to a discount tier:
- _FREE_
- _BRONZE_
- _SILVER_
- _GOLD_
- _PLATINUM_ (Reserved only for enterprise customers)
- _DIAMOND_ (Reserved only for enterprise customers)
```

> When setting pricing for a PPE/PPR Actor, you can define different prices for different tiers.
> While optional, we strongly recommend offering progressively lower prices for higher discount tiers to attract large enterprise customers who may spend thousands or tens of thousands of dollars on your Actor.
> The unit platform usage costs for higher tiers will also be lower, helping you maintain healthy profit margins, as explained in the [Computing your costs for PPE and PPR Actors](#computing-your-costs-for-ppe-and-ppr-actors) section.
> 
> While we don't recommend setting very high prices for free users by default, because it limits the users' ability to evaluate your Actor, we understand that sometimes it's necessary to protect you from fraud or excessive use of your internal APIs, in which case setting a high price for free users might be justified.
> 
> During an Actor run, you can access the user's discount tier through run environment variables or by querying user data via our API. This allows you to potentially provide premium services to users in higher discount tiers.
> 
> Apart from these standard tiers listed above, there are further tiers reserved for enterprise customers, such as `PLATINUM` and `DIAMOND`. If you are interested in providing enterprise level services and attracting large customers, please reach out to us.

Reviewer: Is there any specific channel of communication we want them to reach out to us? Maybe we could take this opportunity and add it here?

```suggestion
You can define different prices for different tiers.
While optional, we recommend offering progressively lower prices for higher discount tiers. This approach can significantly improve attractiveness of your Actor to large enterprise customers who may spend thousands or tens of thousands of dollars on it.

Your platform costs are also lower for these higher tier, which helps maintain healthy profit margins. This is further detailed in the [Computing your costs for PPE and PPR Actors](#computing-your-costs-for-ppe-and-ppr-actors) section.

By default, we advise against setting excessively high prices for _FREE_ tier users, as this can limit the ability to evaluate your Actor thoroughly. However, in certain situations, such as protecting your Actor from fraudulent activity or excessive use of your internal APIs, a higher price for _FREE_ tier users might be justified.

During an Actor run, you can identify the user's discount tier through Actor run environment variables or by querying user data via the Apify API. This capability allows you to offer premium features or differentiated service levels to users in higher discount tiers.

In addition to the standard tiers, Apify provides further tiers specifically for enterprise customers, including _PLATINUM_ and _DIAMOND_ tiers. If you are interested in offering enterprise-level services and attracting major clients, please contact us.
```

> ### Computing your costs for PPE and PPR Actors
> For both PPE and PPR Actors, profit is computed using the formula `(0.8 * revenue) - costs`. In this section, we'll explain how the `costs` component is calculated.
> 
> When paying users run your Actor, it generates platform usage in the form of compute units, data traffic, API operations etc. This usage determines the `costs` in the profit formula above. Note that usage by free users is covered by Apify and doesn't affect your costs. To calculate costs in dollars, multiply the unit cost of each service by the quantity consumed. For example, if a run uses 10 compute units (CUs) at $0.5/CU, your cost would be $5.
> 
> As mentioned in the [How to attract larger customers of PPE and PPR Actors](#how-to-attract-larger-customers-of-ppe-and-ppr-actors) section, the user's discount tier determines the unit costs applied to their runs if the Actor has tiered discounts set up. Higher tiers have lower unit costs, allowing you to offer lower prices to high-tier customers while maintaining healthy profit margins.
> 
> The table below summarizes the unit prices for cost computation across different discount tiers.
> 
> | Service                         | Price is per | `FREE`  | `BRONZE` | `SILVER` | `GOLD`  |
> |:-------------------------------|:--------------|--------:|---------:|---------:|--------:|
> | Compute unit                   | CU            | $0.4    | $0.4     | $0.3     | $0.25   |
> | Residential proxies            | GB            | $8      | $8       | $7.5     | $7      |
> | SERPs proxy                    | 1,000 SERPs   | $2.5    | $2.5     | $2       | $1.7    |
> | Data transfer - external       | GB            | $0.2    | $0.2     | $0.19    | $0.18   |
> | Data transfer - internal       | GB            | $0.05   | $0.05    | $0.045   | $0.04   |
> | Dataset - reads               | 1,000 reads   | $0.0004 | $0.0004  | $0.00036 | $0.00032|
> | Dataset - writes              | 1,000 writes  | $0.005  | $0.005   | $0.0045  | $0.004  |
> | Key-value store - reads       | 1,000 reads   | $0.005  | $0.005   | $0.0045  | $0.004  |
> | Key-value store - writes      | 1,000 writes  | $0.05   | $0.05    | $0.045   | $0.04   |
> | Key-value store - lists       | 1,000 lists   | $0.05   | $0.05    | $0.045   | $0.04   |
> | Request queue - reads         | 1,000 reads   | $0.004  | $0.004   | $0.0036  | $0.0032 |
> | Request queue - writes        | 1,000 writes  | $0.02   | $0.02    | $0.018   | $0.016  |
> 
> If you decide not to offer tiered discounts on your Actor, the unit prices for FREE tier apply. To offer enterprise level services and unlock even cheaper unit prices for enterprise customres, please reach out to us.

Reviewer: ```suggestion
### Computing your costs for PPE and PPR Actors

For both PPE and PPR Actors, profit is computed using the formula `(0.8 * revenue) - costs`. In this section, we'll explain how the `costs` component is calculated.

When paying users run your Actor, it generates platform usage in the form of compute units, data traffic, API operations etc. This usage determines the `costs` in the profit formula above. 

:::note _FREE_ tier usage

Platform usage by _FREE_ tier users is covered by Apify and does not contribute to your costs.

:::

To calculate your costs in dollars for a specific run by paying user, multiply the unit cost of each service by the quantity consumed. For example, if a _BRONZE_ tier user run uses 10 compute units (CUs) at $0.4/CU, your cost would be $4.

As highlighted in the [How to attract larger customers of PPE and PPR Actors](#how-to-attract-larger-customers-of-ppe-and-ppr-actors) section, if your Actor uses tiered pricing, the user's discount tier determines the unit costs applied to their runs. Your costs are lower for higher tiers, enabling you to offer more competitive pricing to these customers, while sustaining healthy profit margins.

The following table summarizes the platform unit costs used for your cost computation  across different discount tiers.

| Service                         | Price is per | _FREE_  | _BRONZE_ | _SILVER_ | _GOLD_  |
|:-------------------------------|:--------------|--------:|---------:|---------:|--------:|
| Compute unit                   | CU            | $0.4    | $0.4     | $0.3     | $0.25   |
| Residential proxies            | GB            | $8      | $8       | $7.5     | $7      |
| SERPs proxy                    | 1,000 SERPs   | $2.5    | $2.5     | $2       | $1.7    |
| Data transfer - external       | GB            | $0.2    | $0.2     | $0.19    | $0.18   |
| Data transfer - internal       | GB            | $0.05   | $0.05    | $0.045   | $0.04   |
| Dataset - reads               | 1,000 reads   | $0.0004 | $0.0004  | $0.00036 | $0.00032|
| Dataset - writes              | 1,000 writes  | $0.005  | $0.005   | $0.0045  | $0.004  |
| Key-value store - reads       | 1,000 reads   | $0.005  | $0.005   | $0.0045  | $0.004  |
| Key-value store - writes      | 1,000 writes  | $0.05   | $0.05    | $0.045   | $0.04   |
| Key-value store - lists       | 1,000 lists   | $0.05   | $0.05    | $0.045   | $0.04   |
| Request queue - reads         | 1,000 reads   | $0.004  | $0.004   | $0.0036  | $0.0032 |
| Request queue - writes        | 1,000 writes  | $0.02   | $0.02    | $0.018   | $0.016  |

If you decide not to offer tiered discounts on your Actor, the unit prices for _FREE_ tier apply. To offer enterprise level services and unlock even cheaper unit prices for enterprise customers, please reach out to us.
```
